plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/plan/AltibaseExecutionPlan.java

```diff
@@ -0,0 +1,178 @@
+        this.query = query;
+        this.rootNodes = nodes;
+    }
+
+    /**
+     * Get Execution Plan by firing query to database. 
+     */
+    public void explain() throws DBException {
+        try {
+            JDBCPreparedStatement dbStat = session.prepareStatement(getQueryString());
+            // Read explained plan
+            try {
+                String plan = getExplainPlan(session, query);
+                rootNodes = AltibasePlanBuilder.build(dataSource, plan);
+            } finally {
+                dbStat.close();
+            }
+        } catch (SQLException e) {
+            throw new DBCException(e, session.getExecutionContext());
+        }
+    }
+
+    @Override
+    public String getQueryString() {
+        return query;
+    }
+
+    @Override
+    public String getPlanQueryString() throws DBException {
+        return planQuery;
+    }
+
+    @Override
+    public List<? extends DBCPlanNode> getPlanNodes(Map<String, Object> options) {
+        return rootNodes;
+    }
+
+    /**
+     *  Get explain plan result string using Java reflection. 
+     */
+    private String getExplainPlan(JDBCSession session, String query) {
+        Statement stmt = null;
+
+        Connection conn = null;
+        Class<? extends Connection> clazz = null;
+        Method method = null;
+        ExplainPlan expPlan = null;
+        
+        try {
+            conn = session.getOriginal();
+            clazz = conn.getClass();
+            
+            /* 
+             * There are two setExplain methods in Connction class: 
+             * The first one's argument is boolean, the second one's argument is byte.
+             * Here, the second method is required.
+             */
+            method = getMethod2SetExplainWithByteArgType(clazz, 
+                    setExplainPlan, AltibaseDataTypeDomain.BYTE.getTypeName().toLowerCase());
+
+            expPlan = AltibaseConstants.ExplainPlan.getByIndex(
+                    dataSource.getContainer().getPreferenceStore().getInt(
+                            AltibaseConstants.PREF_EXPLAIN_PLAN_TYPE));
+
+            method.invoke(conn, expPlan.getArgValue());
+            stmt = conn.prepareStatement(query);
+            
+            // In case of EXPLAIN_PLAN_ON, need to execute query
+            if (expPlan == AltibaseConstants.ExplainPlan.ON) {
+                if (query.trim().toUpperCase().startsWith("SELECT")) {
+                    stmt.getClass().getMethod("executeQuery").invoke(stmt);
+                } else {
+                    stmt.getClass().getMethod("execute").invoke(stmt);
+                }
+            }
+            
+            planQuery = (String) stmt.getClass().getMethod("getExplainPlan").invoke(stmt);
+
+        } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException | NoSuchMethodException
+                | SecurityException | SQLException | ArrayIndexOutOfBoundsException e) {
```

Comment:
Just use a common parent type for these, e.g. `Exception`

Reply:
Thanks for the tip :)